tests/lib/state_test.py

```diff
@@ -241,6 +243,81 @@ def test_absent_cannot_remove_absent(self):
         assert not obj1.is_match(obj2)
 
 
+def test_state_empty_routes():
+    route_state = state.State(
+        {
+            Route.KEY: {
+                Route.CONFIG: []
+            }
+        }
+    )
+
+    assert {} == route_state.config_iface_routes
+
+
+def test_state_iface_routes_with_distinct_ifaces():
+    routes = _get_mixed_test_routes()
+    route_state = state.State(
+        {
+            Route.KEY: {
+                Route.CONFIG: routes
+            }
+        }
+    )
+    expected_indexed_route_state = defaultdict(list)
+    for route in routes:
+        iface_name = route[Route.NEXT_HOP_INTERFACE]
+        expected_indexed_route_state[iface_name].append(route)
+
+    for routes in six.viewvalues(expected_indexed_route_state):
```

Comment:
There seem to be no need for this because there is only one route per interface.